utils/bin/poller/src/main.rs

```diff
@@ -0,0 +1,301 @@
+            println!("Sig: {:?}", sig);
+
+            let offchain_order = Order::try_from(order).unwrap();
+
+            let domain = eip712_domain! {
+                name: "Gnosis Protocol",
+                version: "v2",
+                chain_id: chain_id,
+                verifying_contract: SETTLEMENT.parse()?,
+            };
+
+            let signing_message = offchain_order.eip712_signing_hash(&domain);
+
+            // To do this, we will make use of Multicall3
+            let payload = IMulticall3::tryAggregateCall {
+                requireSuccess: true,
+                calls: vec![
+                    // Inline mapping of pre_interactions to Multicall3::Call
+                    pre_interactions
+                        .iter()
+                        .map(|interaction| IMulticall3::Call {
+                            target: interaction.target,
+                            callData: interaction.callData.clone(),
+                        })
+                        .collect::<Vec<_>>(),
+                    // Inserting the isValidSignature call
+                    vec![IMulticall3::Call {
+                        target: amm,
+                        callData: IERC1271::isValidSignatureCall {
+                            _hash: signing_message,
+                            signature: sig,
+                        }
+                        .abi_encode()
+                        .into(),
+                    }],
+                    // Inline mapping of post_interactions to Multicall3::Call
+                    post_interactions
+                        .iter()
+                        .map(|interaction| IMulticall3::Call {
+                            target: interaction.target,
+                            callData: interaction.callData.clone(),
+                        })
+                        .collect::<Vec<_>>(),
+                    {
+                        match other_post_interactions {
+                            Some(other_post_interactions) => vec![other_post_interactions],
+                            None => vec![],
+                        }
+                    },
+                ]
+                .concat(),
+            }
+            .abi_encode();
+            println!(
+                "\nsimulateDelegateCall payload: {:?}",
+                hex::encode(payload.clone())
+            );
+
+            let settlement = GPv2Settlement::new(SETTLEMENT.parse().unwrap(), provider.clone());
+            let result = settlement
+                .simulateDelegatecall(MUTLICALL3.parse().unwrap(), payload.into())
+                .call()
+                .await;
+
+            match result {
+                Ok(result) => {
+                    let response =
+                        IMulticall3::tryAggregateCall::abi_decode_returns(&result.response, true)?;
+
+                    let length = pre_interactions.len() + post_interactions.len() + 1;
```

Comment:
~This should probably be `payload.calls.len()`, for example `other_post_interactions` is missing.~
This shouldn't be defined so early because it has no context, only when it's needed.

Reply:
Done